### PR TITLE
New function util.WrapText() and updates to util.Ask()

### PR DESCRIPTION
**Pull request by puerco**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:

This PR adds one function to the util package and improves the current `util.Ask()` implementation

The new function `util.WrapText()` takes a long text string and word wraps it to the specified line size

`util.Ask()` has been improved to recognize patterns in the expected answer in order to have more answer choices, expected unsuccessful responses and a default answer to return in case of an empty response from the user.

A simple test is included for `util.WrapText()`

#### Which issue(s) this PR fixes:
Needed by  #1467 

#### Special notes for your reviewer:

Although `util.Ask()` has been improved, the interface is the same.

#### Does this PR introduce a user-facing change?

```release-note
- Added `util.WrapText()` which allows to word wrap text to a specified line size
- Improvements to `util.Ask()` to recognize more choices, invalid expected answers and a default value.
```

### CODE REVIEW

File: pkg/util/common.go [hunk 1]
```diff
@@ -167,6 +167,74 @@ common::askyorn () {
 }
 */
 
+// AskYorN ask a Yes or No question and return a bool with the answer
+func AskYorN(question string) (answer bool, err error) {
+	var response string
+	i := 0
+	for {
```
Comment (saschagrunert) [inline pkg/util/common.go:174]:
```suggestion
	for i := 0; i < 10; i++ {
```

File: pkg/util/common.go [hunk 2]
```diff
@@ -167,6 +167,74 @@ common::askyorn () {
...
+		if response == "" {
+			return defaultAnswer, nil
+		}
+
+		if response == "y" || response == "yes" {
+			return true, nil
+		}
+		if response == "n" || response == "no" {
+			return false, nil
+		}
```
Comment (saschagrunert) [inline pkg/util/common.go:227]: Let's refactor this into a separate function to be re-used by `AskYorN`

**Comment by puerco** [comment]
@justaugustus I saw that one, I mainly wrote these to be able to ask a simple y/n with a default. We are going to ask the user _a lot_ of y/n questions in series like this:
Fix release note NNN? (y/N) [ENTER]
Fix release note NNN? (y/N) [ENTER]
Fix release note NNN? (y/N) [ENTER]
, but I see how I could adapt Ask() `to  one and expect a Y answer. 

Would you prefer tat

**Comment by puerco** [comment]
@saschagrunert yes, I'm about to send the `krel release-notes --fix` PR, since it is a rather big one, I'm sending all changes to the other packages in separate PRs to make the review easier.

**Comment by puerco** [comment]
I've added the ability to capture Ctrl+C when asking y/n. The `AskYotN()` functions will now return an error that can be queried to find out if the user pressed ctrl+c

**Comment by puerco** [comment]
Got it @justaugustus I'll strip the functions and resubmit the PR  :+1:

**Comment by puerco** [comment]
@justaugustus what do you think about approaching this like this last commit: getting rid of the `util.Ask()` function altogether? It is only used twice and the new ones give us a lot more flexibility (like handling cancelling input and providing a default answer).

**Comment by puerco** [comment]
I've dropped the new functions and improved `util.Ask()` keeping the same interface but adding the improvements listed in the pr body:
> util.Ask() has been improved to recognize patterns in the expected answer in order to have more answer choices, expected unsuccessful responses and a default answer to return in case of an empty response from the user.

**Comment by puerco** [comment]
@justaugustus can we lift the hold on this one?

**Pull request merged**